pkg/cmd/pulumi/util_test.go

```diff
@@ -355,3 +356,119 @@ func TestStackLoadOption(t *testing.T) {
 		})
 	}
 }
+
+// TestGetUpdateMetadata tests that the update metadata is correctly populated
+// when running a Pulumi program.
+func TestPulumiCLIMetadata(t *testing.T) {
+	t.Parallel()
+
+	var stringFlag string
+	var intFlag int
+	var boolFlag bool
+
+	cmd := &cobra.Command{
+		Use: "my-cli-command",
+	}
+
+	cmd.PersistentFlags().StringVar(
+		&stringFlag, "name", "",
+		"This is a mock string flag that is set on the command.")
+
+	cmd.PersistentFlags().IntVar(
+		&intFlag, "age", 0,
+		"This is a mock int flag that is set on the command.")
+
+	cmd.PersistentFlags().BoolVar(
+		&boolFlag, "human", false,
+		"This is a mock boolean flag that is set on the command.")
```

Comment:
Seeing as we don't care about the actual values, we can use the non-Var functions and we won't have to pass the `var`s in.

```suggestion
	cmd := &cobra.Command{
		Use: "my-cli-command",
	}

	cmd.PersistentFlags().String(
		"name", "",
		"This is a mock string flag that is set on the command.")

	cmd.PersistentFlags().Int(
		"age", 0,
		"This is a mock int flag that is set on the command.")

	cmd.PersistentFlags().Bool(
		"human", false,
		"This is a mock boolean flag that is set on the command.")
```